### PR TITLE
students and teachers authorization + auth middleware for future tasks

### PR DESCRIPTION
**Pull request by OksanaKozhukh**

### CODE REVIEW

File: src/common/middlewares/auth.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as passport from 'passport';
+import { Unauthorized } from '../../common/exeptions/index';
+import { Request, Response, NextFunction } from 'express';
+
+export default (req: Request, res: Response, next: NextFunction) => {
```
Comment (Alan18081) [inline src/common/middlewares/auth.middleware.ts:5]: Please, use named export

File: src/common/middlewares/create.validation.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import * as Joi from 'joi';
+import { Schema } from 'joi';
+import { Request, Response, NextFunction } from 'express';
+
+export const createValidator = (schema: Schema, key: string = 'body') =>
+    (req: Request, res: Response, next: NextFunction) => {
+    const { error } = Joi.validate(req[key], schema);
+    if (error) {
+
```
Comment (Alan18081) [inline src/common/middlewares/create.validation.ts:9]: Remove useless empty line, please

File: src/common/middlewares/errors.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { Request, Response, NextFunction } from 'express';
+
+export default (err, req: Request, res: Response, next: NextFunction) => {
```
Comment (Alan18081) [inline src/common/middlewares/errors.middleware.ts:3]: Please, use named export instead of default

File: src/modules/groups/groups.model.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import { DataTypes, Model, Sequelize } from 'sequelize';
+import { sequelize } from '../../database';
 
 export class Groups extends Model {
-  public static readonly TableName: string = 'Groups';
+  public static readonly TableName: string = 'groups';
```
Comment (Alan18081) [inline src/modules/groups/groups.model.ts:5]: Please, rename `TableName` to `tableName`

File: src/modules/students/students.service.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Students } from './students.model';
+
+class StudentsService {
+  async findOneByEmail(email: string) {
+    const teacher = await Students.findOne({
+      where: { email }
+    });
+    console.log(teacher);
```
Comment (Alan18081) [inline src/modules/students/students.service.ts:8]: Remove useless `console.log`

File: src/modules/teachers/teachers.service.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { Teachers } from './teachers.model';
+
+class TeachersService {
+  async findOneByEmail(email: string) {
+    const teacher = await Teachers.findOne({
+      where: { email }
+    });
+    console.log(teacher);
```
Comment (Alan18081) [inline src/modules/teachers/teachers.service.ts:8]: Remove useless `console.log`

File: src/server.ts [hunk 7]
```diff
@@ -9,6 +10,18 @@ app.use(express.json());
...
+//authorization
+import passport = require ('passport');
+import errorHandler from './common/middlewares/errors.middleware';
+import { strategy } from './common/passport/auth.strategy';
+passport.use(strategy);
+app.use(errorHandler);
+app.use(passport.initialize());
+import { authRoute } from './modules/auth/auth.routes'; 
+app.use('/auth', new authRoute().router);
+//import jwtAuth from './common/middlewares/auth.middleware';
```
Comment (Alan18081) [inline src/server.ts:22]: Why is this code commented?

File: src/common/middlewares/create.validation.ts (hunk 2 above)
Comment (OksanaKozhukh) [reply src/common/middlewares/create.validation.ts]: done

File: src/common/middlewares/errors.middleware.ts (hunk 3 above)
Comment (OksanaKozhukh) [reply src/common/middlewares/errors.middleware.ts]: done

File: src/modules/groups/groups.model.ts (hunk 4 above)
Comment (OksanaKozhukh) [reply src/modules/groups/groups.model.ts]: done

File: src/modules/students/students.service.ts (hunk 5 above)
Comment (OksanaKozhukh) [reply src/modules/students/students.service.ts]: done

File: src/modules/teachers/teachers.service.ts (hunk 6 above)
Comment (OksanaKozhukh) [reply src/modules/teachers/teachers.service.ts]: done

File: src/server.ts (hunk 7 above)
Comment (OksanaKozhukh) [reply src/server.ts]: done

**Pull request merged**